### PR TITLE
docs: created JSDoc annotation and github action to deploy to gh_pages

### PR DESCRIPTION
**Pull request by tpluscode**

### CODE REVIEW

File: lib/Clownface.js [hunk 1]
```diff
@@ -126,18 +197,39 @@ class Clownface {
     return Clownface.fromContext(context)
   }
 
+  /**
+   * Creates graph pointer to one or more blank node(s)
+   * @param {null|string|string[]} [values] blank node identifiers (uses generated if undefined of null)
```
Comment (tpluscode) [inline lib/Clownface.js:202]: Ah, I realized that all the chain methods also accept Clownface itself. 
Should I change like this where applicable? @bergos 

```suggestion
   * @param {null|string|string[]} [values] blank node identifiers (uses generated if undefined of null)
   * @param {null|string|string[]|Clownface|Clownface[]} [values] blank node identifiers (uses generated if undefined of null)
```

Comment (bergos) [reply lib/Clownface.js:202]: Jup, should be also included

File: lib/Clownface.js [hunk 2]
```diff
@@ -17,11 +20,22 @@ class Clownface {
...
+     * @type {Context[]}
+     * @private
+     */
     this._context = terms.map(term => {
       return new Context({ dataset, graph, value: term, factory: this.factory, namespace: this.namespace })
     })
   }
 
+  /**
+   * Gets the current RDF/JS term or undefined if pointer has
```
Comment (bergos) [inline lib/Clownface.js:35]: something got lost at the end

File: lib/Clownface.js [hunk 3]
```diff
@@ -126,18 +208,39 @@ class Clownface {
     return Clownface.fromContext(context)
   }
 
+  /**
+   * Creates graph pointer to one or more blank node(s)
+   * @param {null|string|string[]|Clownface|Clownface[]} [values] blank node identifiers (uses generated if undefined of null)
```
Comment (vhf) [inline lib/Clownface.js:213]:
```suggestion
   * @param {null|string|string[]|Clownface|Clownface[]} [values] blank node identifiers (generates it when falsy)
```

(typos + I think `''` would also generate an id)

File: lib/Clownface.js [hunk 4]
```diff
@@ -126,18 +208,39 @@ class Clownface {
...
+   * @returns {Clownface}
+   */
   blankNode (values) {
     return this.node(values, { type: 'BlankNode' })
   }
 
+  /**
+   * Creates graph pointer to one or more literal node(s)
```
Comment (vhf) [inline lib/Clownface.js:221]:
```suggestion
   * Creates graph pointer to one or more literal nodes
```

File: lib/Clownface.js [hunk 5]
```diff
@@ -126,18 +208,39 @@ class Clownface {
...
+   * @param {string|string[]|boolean|boolean[]|number|number[]|Literal|Literal[]|Clownface|Clownface[]} values blank node identifiers (uses generated if undefined of null)
```
Comment (vhf) [inline lib/Clownface.js:222]: This one is very similar (`undefined of null` where it should probably be `falsy`) as a previous one but a bit weirder (`values blank node identifiers`)

File: lib/Clownface.js [hunk 6]
```diff
@@ -126,18 +208,39 @@ class Clownface {
...
+   * @param {string|Term} [languageOrDatatype] a language tag string or datatype term
+   * @returns {Clownface}
+   */
   literal (values, languageOrDatatype) {
     return this.node(values, { type: 'Literal', datatype: languageOrDatatype })
   }
 
+  /**
+   * Creates graph pointer to one or more named node(s)
```
Comment (vhf) [inline lib/Clownface.js:231]: Same here

File: lib/Clownface.js [hunk 7]
```diff
@@ -94,18 +138,39 @@ class Clownface {
...
+   * Performs the specified action on every graph pointer
+   * @param {ForEachCallback} callback
+   * @returns {Clownface}
+   */
   forEach (callback) {
     return this.toArray().forEach(callback)
   }
 
+  /**
+   * Calls a defined callback function on each graph pointer, and returns an array that contains the results.
```
Comment (vhf) [inline lib/Clownface.js:169]: Returns a new array with the result of applying `callback` on each graph pointer?

File: lib/Clownface.js [hunk 8]
```diff
@@ -56,10 +90,20 @@ class Clownface {
...
+   * Gets the current context's datasets
+   *
+   * @returns {DatasetCore[]}
+   */
   get datasets () {
     return this._context.map(node => node.dataset).filter(Boolean)
   }
 
+  /**
+   * Creates an iterator which iterates and rdf:List of the current term
```
Comment (vhf) [inline lib/Clownface.js:103]: I don't understand this one

File: lib/Clownface.js [hunk 9]
```diff
@@ -17,11 +20,22 @@ class Clownface {
 
     const terms = (term && toArray(term)) || (value && toArray(value)) || [null]
 
+    /**
+     * The underlying context which references actual node being pointed
```
Comment (vhf) [inline lib/Clownface.js:24]: Not sure what this means, it would probably be worth rephrasing it

**Pull request merged**